drivers/sensor/adi/ad2s1210/ad2s1210.c

```diff
@@ -0,0 +1,904 @@
+	/* Configure fault indication pins as inputs */
+	for (uint8_t i = 0; i < (uint8_t)AD2S1210_FAULT_PIN_MAX_VAL; i++) {
+		if (config->fault_gpios[i].port) {
+			ret = gpio_pin_configure_dt(&config->fault_gpios[i], GPIO_INPUT);
+			if (ret < 0) {
+				LOG_ERR("Could not configure fault GPIO %d (%d)", i, ret);
+				return ret;
+			}
+		}
+	}
+
+	data->mode = AD2S1210_MODE_CONFIG;
+
+	/* Set initial mode to CONFIG */
+	ret = ad2s1210_set_mode_pins(dev, AD2S1210_MODE_CONFIG);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/*
+	 * Check clock frequency is 8 or 10 MHz as these are the only values we
+	 * have in the velocity lookup table.
+	 */
+	switch (config->clock_frequency) {
+	case 8192000:
+		data->clock = AD2S1210_CLOCKIN_8MHZ192;
+		break;
+	case 10240000:
+		data->clock = AD2S1210_CLOCKIN_10MHZ24;
+		break;
+	default:
+		data->clock = AD2S1210_CLOCKIN_UNKNOWN;
+		return -EIO;
+	}
+
+	/* Set resolution based on configuration */
+	if (!data->have_resolution_pins) {
+		/* Convert assigned resolution bits to enum */
+		switch (config->assigned_resolution_bits) {
+		case 10:
+			resolution = AD2S1210_RES_10BIT;
+			break;
+		case 12:
+			resolution = AD2S1210_RES_12BIT;
+			break;
+		case 14:
+			resolution = AD2S1210_RES_14BIT;
+			break;
+		case 16:
+			resolution = AD2S1210_RES_16BIT;
+			break;
+		default:
+			LOG_ERR("Invalid assigned resolution bits: %d",
+				config->assigned_resolution_bits);
+			return -EINVAL;
+		}
+	} else {
+		/* Default to 16 bits when using GPIO control */
+		resolution = AD2S1210_RES_16BIT;
+	}
+
+	/* Set resolution */
+	ret = ad2s1210_set_resolution(dev, resolution);
+	if (ret < 0) {
+		LOG_ERR("Could not set resolution (%d)", ret);
+		return ret;
+	}
+
+	/* Enable hysteresis by default */
+	ret = ad2s1210_set_hysteresis(dev, true);
+	if (ret < 0) {
+		LOG_ERR("Could not set hysteresis");
+		return ret;
+	}
+	return 0;
+}
+
+/** Macro used to initialize one ad2s1210 driver instance */
+#define AD2S1210_INIT(i)                                                                           \
+	static struct ad2s1210_data ad2s1210_data_##i;                                             \
+                                                                                                   \
+	static const struct ad2s1210_config ad2s1210_config_##i = {                                \
+		.spi = SPI_DT_SPEC_INST_GET(i, SPI_WORD_SET(8), 0),                                \
+		.sample_gpio = GPIO_DT_SPEC_INST_GET(i, sample_gpios),                             \
+		.mode_gpios =                                                                      \
+			{                                                                          \
+				GPIO_DT_SPEC_INST_GET_BY_IDX(i, mode_gpios, 0),                    \
+				GPIO_DT_SPEC_INST_GET_BY_IDX(i, mode_gpios, 1),                    \
+			},                                                                         \
+		.reset_gpio = GPIO_DT_SPEC_INST_GET_OR(i, reset_gpios, empty_gpio),                \
```

Comment:
This property is required by bindings.
```suggestion
		.reset_gpio = GPIO_DT_SPEC_INST_GET(i, reset_gpios),                               \
```

Reply:
Right, I'm fixing it now.